cmd/mbop/main.go

```diff
@@ -465,68 +510,95 @@ func entitlements(w http.ResponseWriter, r *http.Request) {
 	fmt.Fprint(w, string(userObj.Entitlements))
 }
 
-func mainHandler(w http.ResponseWriter, r *http.Request) {
+func (m *MBOPServer) mainHandler(w http.ResponseWriter, r *http.Request) {
 	log.Info(fmt.Sprintf("%s %s %s\n", r.RemoteAddr, r.Method, r.URL))
 	switch {
 	case r.URL.Path == "/":
 		statusHandler(w, r)
 	case r.URL.Path == "/v1/users":
-		usersV1(w, r)
+		m.usersV1(w, r)
 	case r.URL.Path == "/v1/jwt":
-		jwtHandler(w, r)
+		m.jwtHandler(w, r)
 	case r.URL.Path == "/v1/auth":
-		authHandler(w, r)
+		m.authHandler(w, r)
 	case r.URL.Path[:12] == "/v1/accounts":
-		usersV1Handler(w, r)
+		m.usersV1Handler(w, r)
 	case r.URL.Path[:12] == "/v2/accounts":
-		usersV2V3Handler(w, r)
+		m.usersV2V3Handler(w, r)
 	case r.URL.Path[:12] == "/v3/accounts":
-		usersV2V3Handler(w, r)
+		m.usersV2V3Handler(w, r)
 	case r.URL.Path == "/api/entitlements/v1/services":
-		entitlements(w, r)
+		m.entitlements(w, r)
 	}
 }
 
-var k *keycloak.KeyCloakClient
 var log logr.Logger
 
-func getMux() *http.ServeMux {
-
-	KEYCLOAK_SERVER := os.Getenv("KEYCLOAK_SERVER")
-	KEYCLOAK_USERNAME := os.Getenv("KEYCLOAK_USERNAME")
-	KEYCLOAK_PASSWORD := os.Getenv("KEYCLOAK_PASSWORD")
-	KEYCLOAK_VERSION := os.Getenv("KEYCLOAK_VERSION")
-	if KEYCLOAK_USERNAME == "" {
-		KEYCLOAK_USERNAME = "admin"
-	}
-	if KEYCLOAK_PASSWORD == "" {
-		KEYCLOAK_PASSWORD = "admin"
-	}
-	if KEYCLOAK_VERSION == "" {
-		KEYCLOAK_VERSION = "11.0.0"
-	}
-
+func (m *MBOPServer) getMux() *http.ServeMux {
 	zapLog, err := zap.NewDevelopment()
 	if err != nil {
 		panic(fmt.Sprintf("who watches the watchmen (%v)?", err))
 	}
 	log = zapr.NewLogger(zapLog)
 
-	key, err := keycloak.NewKeyCloakClient(KEYCLOAK_SERVER, KEYCLOAK_USERNAME, KEYCLOAK_PASSWORD, context.Background(), "master", log, KEYCLOAK_VERSION)
-
-	if err != nil {
-		panic(err)
+	oauthClientConfig := clientcredentials.Config{
+		ClientID:       "admin-cli",
+		ClientSecret:   "",
+		TokenURL:       KEYCLOAK_SERVER + "auth/realms/master/protocol/openid-connect/token",
```

Comment:
this requires a `/`  , the `KEYCLOAK_SERVER` is not expected to have a trailing `/' . 

```suggestion
		TokenURL:       KEYCLOAK_SERVER + "/auth/realms/master/protocol/openid-connect/token",
```